### PR TITLE
feat: Added support for mTLS and the ability to disable default endpoint

### PR DESCRIPTION
**Pull request by anavarreteuk**
## Description
I have added the option to have mutual TLS in the api gateway domain and also the possibility to disable the default endpoint as does not require mutual TLS. 

https://aws.amazon.com/blogs/compute/introducing-mutual-tls-authentication-for-amazon-api-gateway/

## Motivation and Context
It gives the developer the option to enable mTLS and disable the non mTLS default endpoint
Closes https://github.com/terraform-aws-modules/terraform-aws-apigateway-v2/issues/12

## Breaking Changes
It does not introduce any breaking change

## How Has This Been Tested?
Tested in my personal environment

### CODE REVIEW

**Comment by antonbabenko** [comment]
Thanks for the PR. In order to proceed with this, could you please make a new example by copying an existing one and show usage of mTLS feature?

**Comment by davidkarlsen** [comment]
@anavarreteuk seems like all that's left is to add an example? Is this something you intend to do?
I think this addition (mtls) would be very valuable.

File: main.tf [hunk 1]
```diff
@@ -44,6 +45,11 @@ resource "aws_apigatewayv2_domain_name" "this" {
     security_policy = "TLS_1_2"
   }
 
+  mutual_tls_authentication {
```
Comment (antonbabenko) [inline main.tf:48]: Probably `mutual_tls_authentication` should be written with `dynamic "mutual_tls_authentication"` to make this module work in both cases - with mTLS and without it.

File: variables.tf [hunk 2]
```diff
@@ -65,6 +65,12 @@ variable "description" {
   default     = null
 }
 
+variable "disable_execute_api_endpoint" {
+  type        = string
+  default     = false
+  description = "Whether clients can invoke the API by using the default execute-api endpoint. To require that clients use a custom domain name to invoke the API, disable the default endpoint"
```
Comment (antonbabenko) [inline variables.tf:71]: Please reorder these keys: description, type, default. Like on other variables already.

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Please update an example to add a case for mTLS.

File: variables.tf (hunk 2 above)
Comment (davidkarlsen) [reply variables.tf:71]: Maybe you can use my fork instead https://github.com/davidkarlsen/terraform-aws-apigateway-v2/tree/mtls ? It has adressed these issues

**Comment by antonbabenko** [comment]
Fixed by #29 and [v0.15.0](https://github.com/terraform-aws-modules/terraform-aws-apigateway-v2/releases/v0.15.0) has been just released.

**Pull request closed without merging**